returnn/extract_prior.py

```diff
@@ -58,3 +71,333 @@ def run(self):
+        with open(self.out_prior_txt_file.get_path(), "rt") as f:
+            merged_scores = np.loadtxt(f, delimiter=" ")
+
+        with open(self.out_prior_xml_file.get_path(), "wt") as f:
+            f.write(
+                '<?xml version="1.0" encoding="UTF-8"?>\n<vector-f32 size="%d">\n'
+                % len(merged_scores)
+            )
+            f.write(" ".join("%.20e" % s for s in merged_scores) + "\n")
+            f.write("</vector-f32>")
+
+    def plot(self):
+        import matplotlib
+
+        matplotlib.use("Agg")
+        import matplotlib.pyplot as plt
+
+        with open(self.out_prior_txt_file.get_path(), "rt") as f:
+            merged_scores = np.loadtxt(f, delimiter=" ")
+
+        xdata = range(len(merged_scores))
+        plt.semilogy(xdata, np.exp(merged_scores))
+        plt.xlabel("emission idx")
+        plt.ylabel("prior")
+        plt.grid(True)
+        plt.savefig(self.out_prior_png_file.get_path())
+
+    @classmethod
+    def create_returnn_config(
+        cls,
+        model_checkpoint,
+        returnn_config,
+        prior_data,
+        log_verbosity,
+        device,
+        **kwargs,
+    ):
+        """
+        Creates compute_prior RETURNN config
+        :param Checkpoint model_checkpoint:  TF model checkpoint. see `ReturnnTrainingJob`.
+        :param ReturnnConfig returnn_config: object representing RETURNN config
+        :param dict[str]|None prior_data: dataset used to compute prior (None = use one train epoch)
+        :param int log_verbosity: RETURNN log verbosity
+        :param str device: RETURNN device, cpu or gpu
+        :rtype: ReturnnConfig
+        """
+        assert device in ["gpu", "cpu"]
+        original_config = returnn_config.config
+        assert "network" in original_config
+
+        config = {
+            "load": model_checkpoint.ckpt_path,
+        }
+
+        config.update(copy.deepcopy(original_config))  # update with the original config
+
+        # override always
+        config["task"] = "compute_priors"
+
+        if prior_data is not None:
+            config["train"] = {
+                **original_config["train"].copy(),
+                **prior_data,
+            }
+
+        post_config = {
+            "device": device,
+            "log": ["./returnn.log"],
+            "log_verbosity": log_verbosity,
+        }
+
+        post_config.update(copy.deepcopy(returnn_config.post_config))
+
+        res = copy.deepcopy(returnn_config)
+        res.config = config
+        res.post_config = post_config
+        res.check_consistency()
+
+        return res
+
+    @classmethod
+    def hash(cls, kwargs):
+        d = {
+            "returnn_config": ReturnnComputePriorJob.create_returnn_config(**kwargs),
+            "returnn_python_exe": kwargs["returnn_python_exe"],
+            "returnn_root": kwargs["returnn_root"],
+        }
+        return super().hash(d)
+
+
+class ReturnnRasrComputePriorJob(ReturnnComputePriorJob, ReturnnRasrTrainingJob):
```

Comment:
A minimal docstring would be nice at least, something like "Prior job that uses a Rasr Dataset" or so....